src/Features/LanguageServer/ProtocolUnitTests/Ordering/RequestOrderingTests.cs

```diff
@@ -96,38 +96,6 @@ public async Task MutatingDoesntWaitForNonMutating()
             Assert.True(responses[2].StartTime < responses[1].EndTime);
         }
 
-        [Fact]
-        public async Task NonMutatingOperatesOnTheRightSolutions()
```

Comment:
Shouldn't this test still apply?  The initial solution used by the first three requests should be the same, but the third request should return a new solution that is used by the last two requests.

Reply:
Its not that the test doesn't apply, its that it can't be easily validated any more. Originally I left the `UpdateSolution` in place and this still worked, but it felt strange that didChange would construct a changed solution, only for the queue to break it down into documents to track, and then reconstruct another solution again before the next request.

I'll add more tests and see if they cover enough.